### PR TITLE
add one time auth token + add gpt4 trim

### PR DESCRIPTION
**Pull request by liej6799**
currently it will generate one time token, maybe can improve by checking is token valid. Current limitation would be need otp code all the time when login.

### CODE REVIEW

File: openplayground-api/src/openplayground.py [hunk 1]
```diff
@@ -72,4 +72,61 @@ def generate(self, model, prompt, **kwargs):
...
+      yield data
+
+class Auth:
+    # sign in
+    api_url = "https://clerk.nat.dev/v1/client/sign_ins/"
+
+    def __init__(self):
+        self.session = requests.Session()
+        self.headers = {
+            # "Host": "clerk.nat.dev",
```
Comment (ading2210) [inline openplayground-api/src/openplayground.py]: Don't comment out the host and user-agent headers, cause then the user-agent will end up defaulting to `python-requests/2.25.0` or something similar.

File: openplayground-api/src/openplayground.py [hunk 2]
```diff
@@ -72,4 +72,61 @@ def generate(self, model, prompt, **kwargs):
...
+            # "Sec-Fetch-Site": "same-site",
+            "Origin": "https://accounts.nat.dev",
+        }
+        self.session.params={
+            "_clerk_js_version": "4.32.5",
+        },
+
+        self.session.headers.update(self.headers)
+    
+    def login_part_1(self, email_address):
```
Comment (ading2210) [inline openplayground-api/src/openplayground.py:104]: Maybe rename these functions to describe what they're actually doing? Also, I may be mistaken, but it looks like `login_part_1` and `login_part_2` can be combined into the same function.

**Review by ading2210 — CHANGES_REQUESTED** [review]
This looks good, but I'd like to remove the dependency on .env files, since they're only used for the example. Maybe use argparse instead? Perhaps the example script could be invoked like this?
```
python3 example.py --email "sample@example.com"
python3 example.py --token "..."
```

**Comment by liej6799** [comment]
done change from env to argparse

**Review by ading2210 — APPROVED** [review]
Alright, I'm merging these changes.

**Pull request merged**